speechbrain/core.py

```diff
@@ -839,24 +852,29 @@ def fit_batch(self, batch):
         -------
         detached loss
         """
+        should_step = self.step % self.grad_accumulation_factor == 0
         # Managing automatic mixed precision
         if self.auto_mix_prec:
             self.optimizer.zero_grad()
             with torch.cuda.amp.autocast():
                 outputs = self.compute_forward(batch, Stage.TRAIN)
                 loss = self.compute_objectives(outputs, batch, Stage.TRAIN)
-            self.scaler.scale(loss).backward()
-            self.scaler.unscale_(self.optimizer)
-            if self.check_gradients(loss):
-                self.scaler.step(self.optimizer)
-            self.scaler.update()
+            self.scaler.scale(loss/self.grad_accumulation_factor).backward()
```

Comment:
If you might be using DDP you might want to consider no_sync().
https://github.com/pytorch/pytorch/issues/43201

Reply:
Thanks, looks like that would speed things up! @TParcollet I think you've had some recipes using both gradient accumulation and DDP, got time to add this?